### ISSUE TITLE
Add "eval" option to BaseTester.get_all_embeddings

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#91]
```model.eval()``` is called in ```BaseTester.test()```. However, some users might find it convenient to instead use the ```get_all_embeddings``` method, but eval is not called in that method, even though that is what may be expected. (If eval is not called, BatchNorm and Dropout layers are still in training mode.)

@wconnell This might be of interest to you.

**Comment by wconnell** [comment]
This should catch the case when no `embedder_model` is supplied - right now assumes this is passed

**Comment by KevinMusgrave** [comment]
@wconnell Fixed in v0.9.86.dev2
